### PR TITLE
Add fcst dependency to ocnanalprep

### LINKED ISSUE #2723: Ocean analysis prep is failing
### What is wrong?

Wrong job dependency. 

### What should have happened?

Wait for all the necessary files to be present on disk before starting as opposed to just checking for 1.

### What machines are impacted?

All or N/A

### Steps to reproduce

A combination of cycling and bad luck.

### Additional information

Here's the message from @JessicaMeixner-NOAA :

We had a failure in Jiande's runs: /scratch1/NCEPDEV/climate/Jiande.Wang/working/g-w-cycle/cycle/C03/COMROOT/C03/logs/2021080218/gdasocnanalprep.log.0  The issue is: 
 File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/fsutils.py", line 85, in cp
    raise OSError(f"unable to copy {source} to {target}")
OSError: unable to copy /scratch1/NCEPDEV/climate/Jiande.Wang/working/g-w-cycle/cycle/C03/COMROOT/C03/gdas.20210802/12//model_data/ice/restart/[20210802.150000.cice_model.res.nc](http://20210802.150000.cice_model.res.nc/) to /scratch1/NCEPDEV/climate/Jiande.Wang/working/g-w-cycle/cycle/TMP/RUNDIRS/C03/gdasocnanal_18/Data/[20210802.150000.cice_model.res.nc](http://20210802.150000.cice_model.res.nc/)

Cathy's detective work, saw that this looks to be a dependency issue (copying Cathy's chat from another thread): 

ocnanlprep depends on prepoceanobs.  prepoceanobs depends on the existence of previous cycles ocean history f009 file.  looking at the ocean history and the ocean restart timestamps, it looks like the ocean history files finish writing faster than the restarts

[Catherine.Thomas@hfe07 history]$ ls -l ../restart/
total 12357700
-rw-r--r-- 1 Jiande.Wang climate 3855258159 Jun 26 01:57 [20210802.150000.MOM.res.nc](http://20210802.150000.mom.res.nc/)
-rw-r--r-- 1 Jiande.Wang climate 3892610186 Jun 26 01:57 [20210802.150000.MOM.res_1.nc](http://20210802.150000.mom.res_1.nc/)
-rw-r--r-- 1 Jiande.Wang climate 3905057565 Jun 26 01:57 [20210802.150000.MOM.res_2.nc](http://20210802.150000.mom.res_2.nc/)
-rw-r--r-- 1 Jiande.Wang climate 1001184700 Jun 26 01:47 [20210802.150000.MOM.res_3.nc](http://20210802.150000.mom.res_3.nc/)
[Catherine.Thomas@hfe07 history]$ ls -l
total 9210488
-rw-r--r-- 1 Jiande.Wang climate 2357833989 Jun 26 01:46 [gdas.ocean.t12z.inst.f000.nc](http://gdas.ocean.t12z.inst.f000.nc/)
-rw-r--r-- 1 Jiande.Wang climate 2357833989 Jun 26 01:48 [gdas.ocean.t12z.inst.f003.nc](http://gdas.ocean.t12z.inst.f003.nc/)
-rw-r--r-- 1 Jiande.Wang climate 2357833989 Jun 26 01:50 [gdas.ocean.t12z.inst.f006.nc](http://gdas.ocean.t12z.inst.f006.nc/)
-rw-r--r-- 1 Jiande.Wang climate 2357833989 Jun 26 01:51 [gdas.ocean.t12z.inst.f009.nc](http://gdas.ocean.t12z.inst.f009.nc/)

### Do you have a proposed solution?

_No response_

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
# Description

Adds previous cycle's `fcst` as a dependency to `ocnanalprep` to ensure the availability of restart files to the latter. This addresses a seldomly-encountered race condition where `ocnanalprep` fails due to the lack of the files. 

Resolves https://github.com/NOAA-EMC/global-workflow/issues/2723

# Type of change
- Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

1.5 cycle low-res run on Hera

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
FYI @guillaumevernieres

**Review by guillaumevernieres — DISMISSED** [review]
Looks right to me. Thanks @AndrewEichmann-NOAA .

**Review by WalterKolczynski-NOAA — DISMISSED** [review]
Conditionally approved

**Review by CatherineThomas-NOAA — APPROVED** [review]
Looks good. Thanks @AndrewEichmann-NOAA!

File: workflow/rocoto/gfs_tasks.py [hunk 1]
```diff
@@ -659,7 +659,9 @@ def ocnanalprep(self):
         deps = []
         dep_dict = {'type': 'task', 'name': f'{self.cdump}prepoceanobs'}
         deps.append(rocoto.add_dependency(dep_dict))
-        dependencies = rocoto.create_dependency(dep=deps)
+        dep_dict = {'type': 'task', 'name': f'{self.cdump}fcst', 'offset': f"-{timedelta_to_HMS(self._base['cycle_interval'])}"}
```
Comment (aerorahul) [inline workflow/rocoto/gfs_tasks.py]: I think this should be:
```suggestion
        dep_dict = {'type': 'task', 'name': 'gdasfcst', 'offset': f"-{timedelta_to_HMS(self._base['cycle_interval'])}"}
```
When `RUN=gfs`, you will get better backgrounds from the previous `RUN=gdas`.
Also, if you do cycling with `gfs_cyc=1` i.e. run the long forecast at 00z only, there is no `gfsfcst` at 18z and this dependency will never be satisfied.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by DavidHuber-NOAA** [comment]
It looks like the CI tests failed on WCOSS2 due to a missing namelist file.  From `/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2728/RUNTESTS/COMROOT/C96_atmaerosnowDA_c2b1f8c9/logs/2021122100/gfsaeroanlrun.log`:
```
nid001611.dogwood.wcoss2.ncep.noaa.gov 0: OOPS_STATS Run start                                - Runtime:      4.27 sec,  Memory: total:    82.44 Gb, p
er task: min =   183.27 Mb, max =   259.10 Mb
Run: Starting oops::Variational<FV3JEDI, UFO and IODA observations>
OOPS_STATS Variational start                        - Runtime:     4.27 sec,  Local Memory:   201.67 Mb
nid001611.dogwood.wcoss2.ncep.noaa.gov 0:  
FATAL from PE     0: get_ascii_file_num_lines: File ./fv3jedi/fmsmpp.nml does not exist.

nid001611.dogwood.wcoss2.ncep.noaa.gov 0:  
FATAL from PE     0: get_ascii_file_num_lines: File ./fv3jedi/fmsmpp.nml does not exist.

nid001611.dogwood.wcoss2.ncep.noaa.gov 0: Image              PC                Routine            Line        Source                 
libifcoremt.so.5   000014D5A63E0D4A  tracebackqq_          Unknown  Unknown
libfms.so          000014D5AB6D3B63  mpp_mod_mp_mpp_er     Unknown  Unknown
libfms.so          000014D5AB6CB0C8  mpp_mod_mp_get_as     Unknown  Unknown
libfms.so          000014D5AB6CA864  mpp_mod_mp_read_i     Unknown  Unknown
libfms.so          000014D5AB6CFC2C  mpp_mod_mp_mpp_in     Unknown  Unknown
libfms.so          000014D5AB24A019  fms_mod_mp_fms_in     Unknown  Unknown
libfv3jedi.so      000014D5D47FD563  fv3jedi_geom_mod_     Unknown  Unknown
libfv3jedi.so      000014D5D47CEBC0  fv3jedi_geom_init     Unknown  Unknown
libfv3jedi.so      000014D5D4799284  _ZN7fv3jedi8Geome     Unknown  Unknown
gdas.x             000000000059EDCF  Unknown               Unknown  Unknown
gdas.x             00000000007C8006  Unknown               Unknown  Unknown
gdas.x             0000000000619599  Unknown               Unknown  Unknown
gdas.x             0000000000619489  Unknown               Unknown  Unknown
gdas.x             00000000006033D1  Unknown               Unknown  Unknown
gdas.x             000000000060176D  Unknown               Unknown  Unknown
liboops.so         000014D5A9C3C0EB  _ZN4oops3Run7exec     Unknown  Unknown
gdas.x             00000000004D749E  Unknown               Unknown  Unknown
gdas.x             00000000004C563D  Unknown               Unknown  Unknown
libc-2.31.so       000014D5A656724D  __libc_start_main     Unknown  Unknown
gdas.x             00000000004C502A  Unknown               Unknown  Unknown
```

**Comment by AndrewEichmann-NOAA** [comment]
Why would that happen? I don't see how this PR touches that.

**Comment by DavidHuber-NOAA** [comment]
A manual test on WCOSS2 was successful after merging in develop.  Will sync and relaunch WCOSS2 tests.

**Pull request merged**